src/jit/x64/emit.dasc

```diff
@@ -535,6 +536,50 @@ void MVM_jit_emit_primitive(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJ
         | mov WORK[dst], TMP5;
         break;
     }
+    case MVM_OP_getlexreldyn: {
+        MVMint16 dst  = ins->operands[0].reg.orig;
+        MVMint16 ctx  = ins->operands[1].reg.orig;
+        MVMint16 name = ins->operands[2].reg.orig;
+        | mov TMP5, aword WORK[ctx];
+        | cmp_repr_id TMP5, TMP6, MVM_REPR_ID_MVMContext;
+        | je >1;
+        | test_type_object TMP5;
+        | jz >1;
+        | throw_adhoc "getlexreldyn needs a context"
+        |1:
+        | mov TMP6, CONTEXT:TMP5->body.context;
+        | mov ARG1, TC
+        | mov ARG2, aword WORK[name];
+        | mov ARG3, TMP6;
+        | callp &MVM_frame_getdynlex;
+        | mov WORK[dst], RV;
+        break;
+    }
+    case MVM_OP_getlexrelcaller: {
+        MVMint16 dst  = ins->operands[0].reg.orig;
+        MVMint16 ctx  = ins->operands[1].reg.orig;
+        MVMint16 name = ins->operands[2].reg.orig;
+        | mov TMP5, aword WORK[ctx];
+        | cmp_repr_id TMP5, TMP6, MVM_REPR_ID_MVMContext;
+        | je >1;
+        | test_type_object TMP5;
+        | jz >1;
+        | throw_adhoc "getlexrelcaller needs a context";
```

Comment:
Same as another comment I added, this allows both type-object MVMContext and concrete non-MVMObject, and it should not. Apologies for not spotting it earlier.